### PR TITLE
Add map type as MarkupInputType

### PR DESCRIPTION
**Pull request by gabaldon**
This pr adds a map type as MarkupInputType so it could be included as ```Match``` first argument.

### CODE REVIEW

File: src/script.ts [hunk 1]
```diff
@@ -83,8 +82,9 @@ export class Script {
   }
 
   public getOutputType(): OutputType {
-    const lastOperator = this.getLastOperator()
-    return lastOperator ? lastOperator.operatorInfo.outputType : this.firstType
+    return this.getLastOperator() ? this.operators.reduce((acc, operator) => {
+      return operator.operatorInfo.outputType === OutputType.Same ? acc : operator.operatorInfo.outputType
+    }, this.firstType) : this.firstType
```
Comment (Tommytrg) [inline src/script.ts]:
```suggestion
    return this.getLastOperator() ? this.operators.reduce((acc, operator) => {
      let outputType = operator.operatorInfo.outputType
      return outputType === OutputType.Same ? acc : outputType
    }, this.firstType) : this.firstType
```

File: src/utils.ts [hunk 2]
```diff
@@ -106,6 +109,9 @@ export function getDefaultMirArgumentByType(type: MirArgumentType): MirArgument
     case MirArgumentType.String:
       return ''
     case MirArgumentType.Subscript:
+      return [DEFAULT_OPERATOR]
+    //TODO: Return empty object
+    case MirArgumentType.Map:
```
Comment (Tommytrg) [inline src/utils.ts:113]: Should this return an empty object `{ }`?

**Review by Tommytrg — CHANGES_REQUESTED** [review]
Check default value for `map`

File: src/types.ts [hunk 3]
```diff
@@ -124,6 +124,7 @@ export enum MarkupInputType {
   Number = 'number',
   Boolean = 'boolean',
   String = 'string',
+  Map = 'map',
```
Comment (Tommytrg) [inline src/types.ts:127]: Could you add test for Argument with a `map` in `getMarkup`, `getMir` and `getJs`?

**Review by Tommytrg — CHANGES_REQUESTED** [review]
Add tests for Argument class with `map`

**Comment by gabaldon** [comment]
duplicated in #38

**Pull request closed without merging**